### PR TITLE
[4.0] Update manifest cache after testing sampledata plugin installation

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue [https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663](https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663) by @brianteeman , thanks for catching that .

In addition, it very likely also fixes issue #25888 . Please report back when testing if it solves both or only the comment mentioned above.

Thanks to @SharkyKZ for pointing me to the right way to fix it.

### Summary of Changes

This Pull Request (PR) adds the update of manifest cache for the testing sample data plugin to the installation routine for that plugin, which is function `checkTestingSampledata` in file `installation/src/Model/DatabaseModel.php`. It works the same way as manifest cache update of all other core extensions, which happens in function `createTables` in the same file, except that a new installer class is used as suggested by @mbabker , so that there is nothing messed up with the currently used installer instance.

This fixes the PHP notices described in [https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663](https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663), which happen when having the testing sample data plugin available, like is is when using the gihub repository clone and not an unzipped nightly build, and then going to the Extension Manager view.

It seems to fix also the PHP notices described in issue #25888 , which happen under the same conditions as mentioned above but on the Joomla Update Component view, but that's not 100% clear yet. There were different test results regarding that for @SharkyKZ and me.

### Testing Instructions

1. Checkout latest 4.0-dev, do `composer install` and `npm install`.

2. Make a new clean installation into an empty database.

3. After successful installation, login to backend, click away the statistics dialog, go to global config, switch PHP error reporting to maximum or deveopment.

4. Now go to "Manage - Extensions" (or "Extensions - Manage", however is called) view. If you log PHP erros into a log file, check that log file, otherwise have display errors on so check if there is a PHP notice shown on the current page.

Result: See section "Actual result" below, item 1.

5. Now go to the Home Dashboard and click the Joomla Update button to go to the Joomla Update view.

Result: See section "Actual result" below, item 2.

6. Now apply the changes made by this PR, delete configuration.php and the database tables and repeat steps 2 to 5 of the above procedure.

Result: See section "Expected result" below.

### Expected result

No PHP notices.

### Actual result

1. On  "Manage - Extensions" view 2 PHP notices as follows:

> PHP Notice:  Undefined property: stdClass::$version in /home/richard/lamp/public_html/joomla-cms-4.0-dev/administrator/components/com_installer/tmpl/manage/default.php on line 108
> PHP Notice:  Undefined property: stdClass::$version in /home/richard/lamp/public_html/joomla-cms-4.0-dev/administrator/components/com_installer/tmpl/manage/default.php on line 124

This belongs to [https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663](https://github.com/joomla/joomla-cms/pull/25910#issuecomment-522141663).

2. On Joomla Update Component view multiple PHP notices as follows:

> PHP Notice: Trying to get property 'version' of non-object in /home/richard/lamp/public_html/joomla-cms/administrator/components/com_joomlaupdate/Model/UpdateModel.php on line 1357

This belongs to issue #25888 .

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by richard67** [comment]
@SharkyKZ Testing instructions just updated. Could you test?

File: installation/src/Model/DatabaseModel.php [hunk 1]
```diff
@@ -951,7 +952,19 @@ public function checkTestingSampledata($db)
...
+		$installer = new Installer;
+
+		if (!$installer->refreshManifestCache($testingPlugin->extension_id))
+		{
+			Factory::getApplication()->enqueueMessage(
+				Text::sprintf('INSTL_DATABASE_COULD_NOT_REFRESH_MANIFEST_CACHE', $testingPlugin->name),
+				'error'
+			);
+
+			return;
```
Comment (SharkyKZ) [inline installation/src/Model/DatabaseModel.php]: Don't need this return.

Comment (richard67) [reply installation/src/Model/DatabaseModel.php]: You are right of course, how could I not see that. Thanks.

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 0f2e2f02ec7b8d1bedaa2f80706e5650c5c596b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25919">issues.joomla.org/tracker/joomla-cms/25919</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 0f2e2f02ec7b8d1bedaa2f80706e5650c5c596b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25919">issues.joomla.org/tracker/joomla-cms/25919</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25919">issues.joomla.org/tracker/joomla-cms/25919</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**